WordPress/src/main/java/org/xmlrpc/android/XMLRPCUtils.java

```diff
@@ -313,13 +313,25 @@ private static String discoverSelfHostedXmlrpcUrl(String siteUrl, String httpUse
         // add the url as provided by the user
         urlsToTry.add(siteUrl);
 
-        // add a sanitized version of the url
-        final String sanitizedURL = sanitizeSiteUrl(siteUrl);
-        if (!urlsToTry.contains(sanitizedURL)) {
-            urlsToTry.add(sanitizedURL);
+        // add a sanitized version of the http url (if the user didn't specify it)
+        final String sanitizedHttpsURL = sanitizeSiteUrl(siteUrl, true);
+        if (!urlsToTry.contains(sanitizedHttpsURL)) {
+            urlsToTry.add(sanitizedHttpsURL);
+        }
+
+        String appendedXmlrpcUrl = appendXMLRPCPath(sanitizedHttpsURL);
+        if (!urlsToTry.contains(appendedXmlrpcUrl)) {
+            appendedXmlrpcUrl += "?rsd";
```

Comment:
Don't think `?rsd` parameter should be added here. The discovery procedure adds it (if necessary) few lines later.